tests/basic/pure_test.go

```diff
@@ -5002,3 +5004,136 @@ var _ = Describe("{RebootingNodesWhileFADAvolumeCreationInProgressUsingZones}",
+		var isFAaccessible bool
+		testName := "validate-pod-name-in-volume"
+
+		flashArrays, err := GetFADetailsUsed()
+		log.FailOnError(err, "Failed to get FA details from pure.json in the cluster")
+		for _, fa := range flashArrays {
+			if fa.Realm != "" {
+				RealmName = fa.Realm
+				faClient, err = pureutils.PureCreateClientAndConnectRest2_x(fa.MgmtEndPoint, fa.APIToken)
+				if err != nil {
+					log.Errorf("Failed to connect to FA using Mgmt IP [%v]", fa.MgmtEndPoint)
+					continue
+				}
+				isFAaccessible = true
+				break
+			}
+		}
+		if !isFAaccessible {
+			log.FailOnError(fmt.Errorf("No FA with realm found in pure.json"), "No FA with realm found in pure.json")
+		}
+		isRealmExists, err := pureutils.IsFARealmExistsOnMgmtEndpoint(faClient, RealmName)
+		log.FailOnError(err, fmt.Sprintf("Failed to check if realm [%v] exists ", RealmName))
+		if !isRealmExists {
+			log.FailOnError(fmt.Errorf("Realm [%v] is not created in FA", RealmName), "is realm created in FA?")
+		}
+		podNameinSC := "Torpedo-Test" + Inst().InstanceID
+		PodNameinFA := RealmName + "::" + podNameinSC
+
+		stepLog := "Create A pod inside Realm"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			_, err = pureutils.CreatePodinFA(faClient, PodNameinFA)
+			log.FailOnError(err, fmt.Sprintf("Failed to create pod [%v] ", PodNameinFA))
+			isPodExists, err := pureutils.IsPodExistsOnMgmtEndpoint(faClient, PodNameinFA)
+			log.FailOnError(err, fmt.Sprintf("Failed to check if pod [%v] exists ", PodNameinFA))
+			if !isPodExists {
+				log.FailOnError(fmt.Errorf("Pod [%v] is not created in FA", PodNameinFA), "is pod created in FA?")
+			}
+			log.InfoD("Pod [%v] created ", PodNameinFA)
+
+		})
+		stepLog = "Assign the pod name to pure_fa_pod_name in the storage class"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			customConfigAppName := skipTestIfNoRequiredCustomAppConfigFound()
+
+			// save the original custom app configs
+			origCustomAppConfigs = make(map[string]scheduler.AppConfig)
+			for appName, customAppConfig := range Inst().CustomAppConfig {
+				origCustomAppConfigs[appName] = customAppConfig
+			}
+
+			// update the custom app config with pod name
+			Inst().CustomAppConfig[customConfigAppName] = scheduler.AppConfig{
+				PureFaPodName: podNameinSC,
+			}
+
+			log.Infof("JustBeforeEach using Inst().CustomAppConfig = %v", Inst().CustomAppConfig)
+			err = Inst().S.RescanSpecs(Inst().SpecDir, Inst().V.String())
+			log.FailOnError(err, fmt.Sprintf("Failed to rescan specs from %s", Inst().SpecDir))
+
+			context, err := Inst().S.Schedule(testName, scheduler.ScheduleOptions{
+				AppKeys:            Inst().AppList,
+				StorageProvisioner: fmt.Sprintf("%v", portworx.PortworxCsi),
+				Namespace:          testName,
+			})
+			log.FailOnError(err, "Failed to schedule application of %v namespace", testName)
+			contexts = append(contexts, context...)
+			ValidateApplications(contexts)
+
+		})
+		stepLog = "Fetch the pod name from the storage class"
```

Comment:
do you mean "Fetch the pod name from the volume"?

Reply:
Added the logging